projects/plugins/jetpack/extensions/shared/memberships-modal.js

```diff
@@ -0,0 +1,270 @@
+let TB_HEIGHT = 0;
+function show( element ) {
+	element.style.display = 'block';
+	element.style.visibility = 'visible';
+}
+
+//add thickbox to href & area elements that have a class of .thickbox
+function a8c_tb_init( domChunk ) {
+	document.querySelectorAll( domChunk ).forEach( function ( el ) {
+		el.addEventListener(
+			'click',
+			function () {
+				const t = this.title || this.name || null;
+				const a = this.href || this.alt;
+				const g = this.rel || false;
+				a8c_tb_show( t, a, g );
+				this.blur();
+				return false;
+			},
+			false
+		);
+	} );
+}
+
+export const a8c_tb_show = function ( caption, url ) {
+	//function called when the user clicks on a thickbox link
+	document.querySelectorAll( 'body', 'html' ).forEach( el => {
+		el.style.height = '100%';
+		el.style.width = '100%';
+	} );
+	document.querySelector( 'html' ).style.overflow = 'hidden';
+	// try {
+	if ( typeof document.body.style.maxHeight === 'undefined' ) {
+		//if IE 6
+
+		if ( document.getElementById( 'TB_HideSelect' ) === null ) {
+			//iframe to hide select elements in ie6
+			document
+				.querySelector( 'body' )
+				.insertAdjacentHTML(
+					'beforeend',
+					"<iframe id='TB_HideSelect'></iframe><div id='TB_overlay'></div><div id='TB_window'></div>"
+				);
+			document.getElementById( 'TB_overlay' ).addEventListener( 'click', a8c_tb_remove, false );
+		}
+	} else if ( document.getElementById( 'TB_overlay' ) === null ) {
+		document
+			.querySelector( 'body' )
+			.insertAdjacentHTML( 'beforeend', "<div id='TB_overlay'></div><div id='TB_window'></div>" );
+		document.getElementById( 'TB_overlay' ).addEventListener( 'click', a8c_tb_remove, false );
+	}
+
+	if ( a8c_tb_detectMacXFF() ) {
+		document.getElementById( 'TB_overlay' ).classList.add( 'TB_overlayMacFFBGHack' ); //use png overlay so hide flash
+	} else {
+		document.getElementById( 'TB_overlay' ).classList.add( 'TB_overlayBG' ); //use background and opacity
+	}
+
+	if ( caption === null ) {
+		caption = '';
+	}
+
+	//code to show html
+	const queryString = url.replace( /^[^?]+\??/, '' );
+	const params = a8c_tb_parseQuery( queryString );
+	TB_WIDTH = params.width * 1 + 30 || 630; //defaults to 630 if no paramaters were added to URL
+	TB_HEIGHT = params.height * 1 + 40 || 440; //defaults to 440 if no paramaters were added to URL
+	const ajaxContentW = TB_WIDTH - 30;
+	const ajaxContentH = TB_HEIGHT - 45;
+
+	if ( url.indexOf( 'TB_iframe' ) !== -1 ) {
+		// either iframe or ajax window
+		const urlNoQuery = url.split( 'TB_' );
+		document.getElementById( 'TB_iframeContent' )?.remove();
+		if ( params.modal !== 'true' ) {
+			//iframe no modal
+			document
+				.getElementById( 'TB_window' )
+				.insertAdjacentHTML(
+					'beforeend',
+					"<div id='TB_title'><div id='TB_ajaxWindowTitle'>" +
+						caption +
+						"</div><div id='TB_closeAjaxWindow'><a href='#' id='TB_closeWindowButton' title='Close'>close</a> or Esc Key</div></div><iframe frameborder='0' hspace='0' src='" +
```

Comment:
It would be nice if those strings were translated. Since you placed the lib in the `extensions` dir, it gets transpiled so you should be able to either:

1. Import `__` from @wordpress/i18n` and translate in place. I would not recommend this since it would bring a lot of extra dependencies on a file that we also load on the frontend.
2. Pass those translations from php to js with `wp_localize_script`. We do not have a specific script handle to attach it to at the moment though.

This all makes me wonder if that new lib should be handled like a real lib, like some of the ones we have in `projects/plugins/jetpack/_inc`. We would then register it as a new script (`jetpack-thickbox`) in `Jetpack::register_assets()`, and pass the translations there. We could then add the script as a dependency when needed.

To give you a better idea of what I mean, I pushed 79d3e769a8ac4b3a89ab4fff209c681d5aa38ee2 in a test branch. Let me know what you think about the idea.

Reply:
That gos with my comment [here](https://github.com/Automattic/jetpack/pull/28318#discussion_r1085052032).

I do like the idea, but maybe we should wrap this PR first and then extend its functionality and share it for other extensions after talking with the team?

Reply:
I opened the discussion here:
p9dueE-6Cp-p2